src/Evolution/Systems/NewtonianEuler/InitializeSourceTerm.hpp

```diff
@@ -0,0 +1,76 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <tuple>
+#include <utility>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "Evolution/Initialization/Tags.hpp"
+#include "Evolution/TypeTraits.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "ParallelAlgorithms/Initialization/MergeIntoDataBox.hpp"
+#include "Utilities/Requires.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "Utilities/TypeTraits.hpp"
+
+namespace NewtonianEuler {
+namespace Actions {
+/// \ingroup InitializationGroup
+///
+/// Uses: nothing
+///
+/// DataBox changes:
+/// - Adds:
+///   * Metavariables::source_term_tag
+///
+/// - Removes: nothing
+/// - Modifies: nothing
+struct InitializeSourceTerm {
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent,
+            Requires<tmpl::list_contains_v<
+                typename db::DataBox<DbTagsList>::simple_item_tags,
+                Initialization::Tags::InitialTime>> = nullptr>
+  static auto apply(db::DataBox<DbTagsList>& box,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ArrayIndex& /*array_index*/, ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    using simple_tags =
+        db::AddSimpleTags<typename Metavariables::source_term_tag>;
+    using compute_tags = db::AddComputeTags<>;
+    auto source_term =
+        Parallel::get<typename Metavariables::initial_data_tag>(cache)
+            .source_term();
```

Comment:
Would it make sense to have the source term in the cache along with the analytic solution?

Reply:
Yes, I think that makes sense

Reply:
ping

Reply:
Most of the ideas devoted to this change would require some changes beyond the scope of this PR, and they would also be applicable to putting EoSs (and possibly other classes) in the global cache. These changes would certainly imply further group discussion, so, after discussing this with @kidder, I opted to leave the source in the databox for now and come back with an enhancement once we know how we want to do it with EoSs.

Reply:
Is there more needed for this than
```c++
template <typename SolutionType>
struct SourceTerm : SourceTermBase, db::SimpleTag {
  using type = typename SolutionType::source_term_type;
  using option_tags = tmpl::list<::OptionTags::AnalyticSolution<SolutionType>>;
  static type create_from_options(
      const SolutionType& analytic_solution) noexcept {
    return analytic_solution.sources();
  }
};
```
and changing some template parameters around?

Reply:
Yea, I think that should work. Maybe we should talk about how to implement this after the exec call? I think EoSs could be done the same or in a similar way.

Reply:
That was my first attempt, but the initial data is not restricted to analytic solutions, or even to analytic data. Adding an `OptionTags` for a generic `InitialData` type might do it, but the very fact of creating the source (or the EoS) as a member of another class seems ultimately suboptimal in cases where you don't really need it to be a member (i.e., when the source doesn't depend on parameters that needed for the initial data, e.g. for numerical initial data.) I'm fine with implementing your suggestion if you think that it will make things easier to update in the future, though.

Reply:
I'd be happy to discuss what's required for (and volunteer to implement) the changes for EoSs.

Reply:
So after our discussion today, I implemented Will's suggestion (last fixup commit). The initialization source term commit will be dropped once I squash.